cpp/src/parquet/encryption/dataset_encryption_config.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "parquet/encryption/crypto_factory.h"
+#include "parquet/encryption/encryption.h"
+#include "parquet/encryption/kms_client.h"
+#include "arrow/dataset/type_fwd.h"
+
+namespace parquet {
+namespace encryption {
+
+struct PARQUET_EXPORT DatasetEncryptionConfiguration {
```

Comment:
Is it better not to expose dataset concepts within the parquet module? For example the line 6 looks strange to me. `ARROW_PARQUET` can be built without `ARROW_DATASET` enabled. We may add a translation layer to pass these config to parquet writer.